File: src/main/java/me/clickism/clicksigns/item/ModItems.java
package me.clickism.clicksigns.item;

import me.clickism.clicksigns.ClickSigns;
import net.minecraft.item.Item;
import net.minecraft.registry.Registries;
import net.minecraft.registry.Registry;

public class ModItems {

    private static Item registerItem(String id, Item item) {
        return Registry.register(Registries.ITEM, ClickSigns.identifier(id), item);
    }

    public static void registerItems() {
    }
}

File: src/main/kotlin/de/clickism/clicksigns/registry/ModItems.kt
package de.clickism.clicksigns.registry

import de.clickism.clicksigns.platform.Platform
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.Blocks
import net.minecraft.world.level.block.state.BlockBehaviour

object ModItems {
    val ROAD_SIGN = Platform.registerBlockWithItem(
        "road_sign",
        BlockBehaviour.Properties.ofFullCopy(Blocks.STONE)
    ) { blockSettings ->
        Block(blockSettings)
    }

    fun initialize() {

    }
}